### ISSUE TITLE
Range information is inconsistent for Program nodes with leading whitespace

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/espree#349]
```js
// espree 3.4.3

require('espree').parse('/* foo */ bar /* baz */', { range: true })
```

<details>
<summary>Output</summary>

```json
{
  "type": "Program",
  "start": 0,
  "end": 23,
  "range": [
    10,
    13
  ],
  "body": [
    {
      "type": "ExpressionStatement",
      "start": 10,
      "end": 13,
      "range": [
        10,
        13
      ],
      "expression": {
        "type": "Identifier",
        "start": 10,
        "end": 13,
        "range": [
          10,
          13
        ],
        "name": "bar"
      }
    }
  ],
  "sourceType": "script"
}
```
</details>

The `Program` node has `start: 0` and `end: 23`, but its `range` is `[10, 13]`.

This happens because espree [transforms the `range` property](https://github.com/eslint/espree/blob/d4bdcb60a01ca3988e9c5aebb1036d7fc304386c/espree.js#L772-L782) to match Esprima's behavior. The transformation is correct, but we should also update the `start` and `end` properties, since they're equivalent in all other cases.

**Comment by JamesHenry** [comment]
Ah I didn't originally see this issue, but see mine in babylon for more comparisons:

https://github.com/babel/babylon/issues/673

**Comment by platinumazure** [comment]
@not-an-aardvark Is this still an issue?

**Comment by fisker** [comment]
Is this going to get fixed?

**Comment by nzakas** [comment]
@fisker can you confirm that this is still an issue?

**Comment by fisker** [comment]
Yes, I'm sure, I found this  during working on Prettier,  `flow` and `espree` are different from other parsers, so I filed issue on flow https://github.com/facebook/flow/issues/8537, and comment here.

**Comment by mdjermanovic** [comment]
`TemplateElement` nodes also have `start` and `end` different from `range`.

**Comment by nzakas** [comment]
This is already marked as accepted, so any volunteers? 😄

**Comment by anikethsaha** [comment]
I am working on this. This should probably be a breaking change.
